### PR TITLE
Added nanos to timestamp when converting

### PR DESCRIPTION
**Pull request by ankur12-1610**
Added `Nanos` to the `timestamp` in `convertSummary` function
Fixes: #1124

### CODE REVIEW

**Comment by ankur12-1610** [comment]
@michelle192837 can you please review?

File: pkg/api/v1/summary.go [hunk 1]
```diff
@@ -54,9 +54,11 @@ func convertSummary(tabSummary *summarypb.DashboardTabSummary) *apipb.TabSummary
 		DetailedStatusMessage: tabSummary.Status,
 		LastRunTimestamp: &timestamp.Timestamp{
 			Seconds: int64(tabSummary.LastRunTimestamp),
+			Nanos:   int32(tabSummary.LastRunTimestamp),
```
Comment (sultan-duisenbay) [inline pkg/api/v1/summary.go:57]: @ankur12-1610 the idea is to convert the fractional part of the `LastRunTimestamp`. We are currently just rounding it off to fetch seconds but discarding the fractional part.

Comment (ankur12-1610) [reply pkg/api/v1/summary.go]: > @ankur12-1610 the idea is to convert the fractional part of the `LastRunTimestamp` coming from `tabSummary` (it is a double there - hence might contain some fractional part of timestamp). We are currently just rounding it off to fetch seconds but discarding the fractional part.
> 
> The fractional part might be useful and needs to be added in nanos to timestamp struct. Once done, the test cases need to be updated too to verify the correctness. Thanks!

Ah! got it!
I've updated the code and used `modf()` for that, I haven't written any tests will do so once the current implemented method is confirmed. Can you please verify?

**Review by sultan-duisenbay — COMMENTED** [review]
LGTM! Please add the test cases now :) - thanks!

File: pkg/api/v1/summary.go (hunk 1 above)
Comment (sultan-duisenbay) [reply pkg/api/v1/summary.go]: looks good to me @ankur12-1610 ! thanks!

**Comment by ankur12-1610** [comment]
> LGTM! Please add the test cases now :) - thanks!

Hey, I've added tests, should I create a different test for Nanos or I should insert it in existing tests as I've done in the recent commit?

**Review by sultan-duisenbay — COMMENTED** [review]
Would be curios to see additional test cases with a variable number of digits in the fractional part (.354 or even .1). Could you please add those too @ankur12-1610 ?

File: pkg/api/v1/summary_test.go [hunk 2]
```diff
@@ -500,9 +508,11 @@ func TestListTabSummariesHTTP(t *testing.T) {
 						LatestPassingBuild:    "Lantern",
 						LastUpdateTimestamp: &timestamp.Timestamp{
 							Seconds: 916166800,
+							Nanos:   0,
```
Comment (sultan-duisenbay) [inline pkg/api/v1/summary_test.go:511]: the default value for number fields in Go is 0, so you don't have specify in that case

Comment (ankur12-1610) [reply pkg/api/v1/summary_test.go]: Got it!

**Comment by ankur12-1610** [comment]
> Would be curios to see additional test cases with a variable number of digits in the fractional part (.354 or even .1). Could you please add those too @ankur12-1610 ?

@sultan-duisenbay should these be enough?

**Review by sultan-duisenbay — CHANGES_REQUESTED** [review]
Sorry for the long turnaround time @ankur12-1610 . Today is a day off in the US and Canada.

I believe you might have missed a typo in one of your test cases. I would advise to run the tests locally first (much faster that way) using `blaze test //...` command

File: pkg/api/v1/summary_test.go [hunk 3]
```diff
@@ -103,18 +111,36 @@ func TestListTabSummaries(t *testing.T) {
...
 							LastRunTimestamp:    float64(916166800),
 						},
+						{
+							DashboardName:       "Marco",
+							DashboardTabName:    "polo-4",
+							Status:              "1/7 tests are failing!",
+							OverallStatus:       summarypb.DashboardTabSummary_ACCEPTABLE,
+							LatestGreen:         "Lantern",
+							LastUpdateTimestamp: float64(0.916166782),
+							LastRunTimestamp:   915166800.9161667820.916166782),
```
[the remark this answers is not in the document]
Comment (ankur12-1610) [reply pkg/api/v1/summary_test.go]: Ah! I forgot to update the code :sweat_smile:
Ran `bazel test //...` and tests related to this file were all **passing**

**Comment by ankur12-1610** [comment]
> Sorry for the long turnaround time @ankur12-1610 . Today is a day off in the US and Canada.
> 
> I believe you might have missed a typo in one of your test cases. I would advise to run the tests locally first (much faster that way) using `blaze test //...` command

Np :smile: 
Ah! I forgot to update the code :sweat_smile:
Ran `bazel test //...` and tests related to this file were all **passing**

**Review by sultan-duisenbay — APPROVED** [review]
LGTM!

**Pull request merged**